dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/mtls/mutliple/multipleapp/KafkaMtlsMultipleAppsTest.java

```diff
@@ -0,0 +1,163 @@
+
+        // The Connection properties for the KafkaTestClient should include all certificates, so it uses the truststore as both trust and key stores so it has both client certs
+        PropertiesAsset appConfig1 = new PropertiesAsset()
+                .addProperty(KafkaTestClientProvider.CONNECTION_PROPERTIES_KEY, KafkaTestClientProvider.encodeProperties(MtlsMultipleKeyStoresTests.connectionProperties()))
+                .include(incomingProperties)
+                .include(outgoingProperties)
+                .include(connectorProperties);
+
+        PropertiesAsset appConfig2 = new PropertiesAsset()
+                .addProperty(KafkaTestClientProvider.CONNECTION_PROPERTIES_KEY, KafkaTestClientProvider.encodeProperties(MtlsMultipleKeyStoresTests.connectionProperties()))
+                .include(incomingProperties2)
+                .include(outgoingProperties2)
+                .include(connectorProperties);
+
+        WebArchive war1 = ShrinkWrap.create(WebArchive.class, APP1_NAME + ".war")
+                .addAsLibraries(kafkaClientLibs())
+                .addAsManifestResource(kafkaPermissions(), "permissions.xml")
+                .addClasses(KafkaMtlsTestServletOne.class, MessagingBeanOne.class)
+                .addPackage(AbstractKafkaTestServlet.class.getPackage())
+                .addPackage(KafkaTestConstants.class.getPackage())
+                .addAsResource(appConfig1, "META-INF/microprofile-config.properties");
+
+        WebArchive war2 = ShrinkWrap.create(WebArchive.class, APP2_NAME + ".war")
+                .addAsLibraries(kafkaClientLibs())
+                .addAsManifestResource(kafkaPermissions(), "permissions.xml")
+                .addClasses(KafkaMtlsTestServletTwo.class, MessagingBeanTwo.class)
+                .addPackage(AbstractKafkaTestServlet.class.getPackage())
+                .addPackage(KafkaTestConstants.class.getPackage())
+                .addAsResource(appConfig2, "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, war1, SERVER_ONLY);
+        ShrinkHelper.exportDropinAppToServer(server, war2, SERVER_ONLY);
+
+        KafkaUtils.copyTrustStore(MtlsMultipleKeyStoresTests.kafkaContainer, server);
+        KafkaUtils.copyKeyStoresToServer(MtlsMultipleKeyStoresTests.kafkaContainer, server);
+
+        server.startServer();
+    }
+
+    @Test
+    public void testMutlipleMtlsApps() throws Exception {
+        runTest(server, APP1_NAME + "/KafkaMtlsServletOne", "putMessages");
+        runTest(server, APP2_NAME + "/KafkaMtlsServletTwo", "putMessages");
+
+        // the Kafka JMX issues are reported as Errors, not expcetions/ffdcs or have codes
+        // so is not picked up on shutdown of the server, meaning that it passes
+        assertNull("Unexpected Kafka Error",
+                server.waitForStringInLog("E Error.*kafka", 1000));
```

Comment:
Since you've spotted this, we should probably wrap up this logic and use it in every test - ideally checking for error messages from Kafka after the server has stopped so we don't need to wait.

Could be done later.

Reply:
If we do move this, we can also run the tests as regular test servlets, rather than calling both servlets here.

Reply:
[#27120](https://github.com/OpenLiberty/open-liberty/issues/27120) has been opened to address the kafka logging issue